### PR TITLE
tools/cmake: fix download url with make variables

### PR DESCRIPTION
**Pull request by mcprat**
Use a make variable pattern for the url
so that only one version number needs to be changed when version is bumped.

Signed-off-by: Michael Pratt <mcpratt@pm.me>

### CODE REVIEW

File: tools/cmake/Makefile [hunk 1]
```diff
@@ -8,12 +8,13 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=cmake
 PKG_VERSION:=3.24.1
+PKG_VERSION_MAJOR:=$(word 1,$(subst ., ,$(PKG_VERSION))).$(word 2,$(subst ., ,$(PKG_VERSION)))
```
Comment (neheb) [inline tools/cmake/Makefile:11]: This is wonderful. A lot of packages in the packages feed can take advantage of this.

Comment (Ansuel) [reply tools/cmake/Makefile:11]: Wonder... should we make this an helper?

Comment (neheb) [reply tools/cmake/Makefile:11]: Probably not.

**Comment by Ansuel** [comment]
@mpratt14 I wonder if you can rebase this on top of master so this gets flagged as merged. You disabled maintainers force push so I can't do it and the only option is to manually merge and flag this as closed :(

**Comment by mpratt14** [comment]
Isn't there a reason that PRs are never merged through github, and instead always manually? something like, github is a one-way mirror and then fast-forward from git.openwrt.org would fail?

that's the reason they added the review block for merging "At least 6 approving reviews are required by reviewers with write access"?? or is it just to prevent accidental misclicks or something

**Comment by Ansuel** [comment]
@mpratt14 yes but with extra step they can still be flagged as merged as you can see some closed pr recently

The extra step needs to rebase the pr on top of master so they can be merged with -ff-only. Normally I would do this for the user by force pushing it but is not possible if the user block this feature for the pr.

**Comment by mpratt14** [comment]
got it 👍🏼

**Comment by mpratt14** [comment]
and by the way, the reason is that I update all my branches together with `git push --mirror origin` so anyone else pushing my branch would get overwritten and that can cause confusion if I'm working on something else while you are working on this

**Comment by Ansuel** [comment]
@mpratt14 see merged :D anyway it's ok it was just to clear why force push is needed and why it's problematic. Everyone has it's reason it's just a pitty to close pr without giving actual credits that they were merged.

**Pull request merged**